pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go

```diff
@@ -123,9 +123,7 @@ func (c *defaultControllerRegistrationSeedControl) Reconcile(obj *gardencorev1be
 	if err != nil {
 		return err
 	}
-	shootList, err := c.k8sGardenClient.GardenCore().CoreV1beta1().Shoots(metav1.NamespaceAll).List(metav1.ListOptions{
-		FieldSelector: fields.SelectorFromSet(fields.Set{core.ShootSeedName: seed.Name}).String(),
```

Comment:
To optimize the list call, does it make sense to have a field selector for `.status.seedName` as well, and use it in conjunction with this one? Is this possible in the first place?

Reply:
Yes I was thinking the same, but couldn't find a way to use `OR` for the field selectors in case the `shoot.spec.seedName` is set to `Target Seed`, but still have a `status.seedName` equal to `Source Seed`. I had another thought - to use the `status.seedName` only, now when we set the status.seedName in the beginning of the creation/reconciliation.

Reply:
Hm sure, could work. Generally, filtering here would help a lot given the large number of shoots in a system.

Reply:
I tried with `FilterFunc` of type `status.seedName=gcp` and this will block the first shoot creation on a new seed because there is a check if extensions are ready :/ before starting the reconciliation. Maybe we can list twice and merge the lists?

Reply:
Hm yeah, sounds reasonable. Thanks for your efforts.